### ISSUE TITLE
Sample not working with 2.4.0 (RELEASE)

### ISSUE THREAD

**Issue by SeaLife** [spring-projects-experimental/spring-authorization-server#154]
**Describe the bug**
The dependency in v0.0.3 in combination with Spring Boot 2.4.0.RELEASE using the example code throws an Error:

```
Caused by: java.lang.IllegalStateException: Found WebSecurityConfigurerAdapter as well as SecurityFilterChain. Please select just one.
```

**To Reproduce**
Create a basic Spring Boot project and add the example configuration:

```java
@Configuration
@EnableWebSecurity
@Import(OAuth2AuthorizationServerConfiguration.class)
public class AuthorizationServerConfiguration {

    // @formatter:off
    @Bean
    public RegisteredClientRepository registeredClientRepository() {
        RegisteredClient registeredClient = RegisteredClient.withId(UUID.randomUUID().toString())
                .clientId("messaging-client")
                .clientSecret("secret")
                .clientAuthenticationMethod(ClientAuthenticationMethod.BASIC)
                .authorizationGrantType(AuthorizationGrantType.AUTHORIZATION_CODE)
                .authorizationGrantType(AuthorizationGrantType.CLIENT_CREDENTIALS)
                .redirectUri("https://oauth.pstmn.io/v1/callback")
                .scope("message.read")
                .scope("message.write")
                .clientSettings(clientSettings -> clientSettings.requireUserConsent(true))
                .build();
        return new InMemoryRegisteredClientRepository(registeredClient);
    }

    // @formatter:on
    @Bean
    public KeyManager keySource() {
        return new StaticKeyGeneratingKeyManager();
    }

    // @formatter:off
    @Bean
    public UserDetailsService users() {
        UserDetails user = User.withDefaultPasswordEncoder()
                .username("user1")
                .password("password")
                .roles("USER")
                .build();
        return new InMemoryUserDetailsManager(user);
    }
    // @formatter:on
}
```

Run the service, it wont come up.

**Expected behavior**
A clean boot just like the 2.4.0-M3 did.

**Sample**

A link to a GitHub repository with a [minimal, reproducible sample](https://stackoverflow.com/help/minimal-reproducible-example).

https://github.com/SeaLife/spring-boot-oauth2-bug-154

**Comment by lspil** [comment]
Hi. I can also reproduce it. As a workaround, you can add yo your config class "extends WebSecurityConfigurerAdapter".

**Comment by SeaLife** [comment]
I renamed the issue, the library/authorization server is working fine so far with 2.4.0 but with the "workaround" @lspil suggested. :)

**Comment by jgrandja** [comment]
Thanks for the report @SeaLife ! This is now resolved via #163
